tests/components/energyzero/test_services.py

```diff
@@ -0,0 +1,55 @@
+"""Tests for the services provided by the EnergyZero integration."""
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.energyzero.const import DOMAIN
+from homeassistant.components.energyzero.services import (
+    ENERGY_SERVICE_NAME,
+    GAS_SERVICE_NAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+
+
+@pytest.mark.usefixtures("init_integration")
+async def test_has_services(
+    hass: HomeAssistant,
+) -> None:
+    """Test the existence of the EnergyZero Service."""
+    assert hass.services.has_service(DOMAIN, GAS_SERVICE_NAME)
+    assert hass.services.has_service(DOMAIN, ENERGY_SERVICE_NAME)
+
+
+@pytest.mark.usefixtures("init_integration")
+@pytest.mark.parametrize("service", [GAS_SERVICE_NAME, ENERGY_SERVICE_NAME])
+@pytest.mark.parametrize("incl_vat", [{"incl_vat": False}, {"incl_vat": True}])
+@pytest.mark.parametrize(
+    "start", [{"start": "2023-01-01 00:00:00"}, {"start": "incorrect date"}, {}]
+)
+@pytest.mark.parametrize(
+    "end", [{"end": "2023-01-01 00:00:00"}, {"end": "incorrect date"}, {}]
+)
+async def test_service(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    service: str,
+    incl_vat: dict[str, bool],
+    start: dict[str, str],
+    end: dict[str, str],
+) -> None:
+    """Test the EnergyZero Service."""
+
+    data = incl_vat | start | end
+
+    try:
+        response = await hass.services.async_call(
+            DOMAIN,
+            service,
+            data,
+            blocking=True,
+            return_response=True,
+        )
+        assert response == snapshot
+    except ServiceValidationError as e:
```

Comment:
Please separate tests as suggested in https://github.com/home-assistant/core/pull/105175#discussion_r1429760216